elbepack/makofiles/preferences.mako

```diff
@@ -6,6 +6,35 @@
 <%!
     import textwrap
     from textwrap import dedent
+
+    def pkg_properties_specified(attrib):
+        """Check if any package pinning properties are provided."""
+        for key in ["version", "origin", "release-archive", "release-component", "release-label", "release-name", "release-origin", "release-version"]:
+            if key in attrib:
+                return True
+        return False
+
+    def generate_pin_text(attrib):
+        """Generate the package pinning properties based on the provided attributes."""
+        # return early as the attributes are mutually exclusive
+        if "version" in attrib:
+            return f'Pin: version {attrib["version"]}'
+
+        if "origin" in attrib:
+            return f'Pin: origin {attrib["origin"]}'
+
+        opts = {
+            'a': attrib.get("release-archive"),
+            'c': attrib.get("release-component"),
+            'l': attrib.get("release-label"),
+            'n': attrib.get("release-name"),
+            'o': attrib.get("release-origin"),
+            'v': attrib.get("release-version"),
+        }
+        release_opts = ', '.join([f'{k}={v}' for k, v in opts.items() if v is not None])
+        if release_opts:
+            return f'Pin: release {release_opts}'
+        return ''
```

Comment:
To me reaching this place looks like an error.
Could you throw a runtime error?

(If it fails during testing I can add the empty return again)

Reply:
Fine by me, added a RuntimeError.